### PR TITLE
🐛 Fix handling binary data type for Mssql source

### LINKED ISSUE #4270: Source MSSQL: Connector provides wrong values for some datatypes
Found by comprehensive tests #3562

## Expected Behavior
A connector should transform source data into an Airbyte datatype without value/meaning losses.

## Current Behavior
General source implementation:
- 'binary', 'varbinary'. Returns binary value instead of actual value.

## Steps to Reproduce
Short way:
1. Checkout master branch
2. Go to `MssqlSourceComprehensiveTest.java` and find tests with appropriate types. Use example values from comment to insert to addInsertValues and addExpectedValues
4. Run the test.
5. :bomb: 
NOTES: To REMEMBER while fixing this bug: Most of the cases would need to be corrected at JdbcUtils, but this is based class that is used by all others JDBC DBs, i.e. all other JDBC DBs will be affected

## Severity of the bug for you
High

## Airbyte Version
0.24.7-alpha

## Connector Version
0.3.7

Origin ticket: https://github.com/airbytehq/airbyte/issues/3794

### PR DESCRIPTION
**Pull request by sashaNeshcheret**
## What
Handled binary abd var binary data types for  Mssql source and fixed appropriate test cases.

## How
We created MssqlOperations class and overridden putBinary method.

## Recommended reading order
1. `x.java`
2. `y.json`
3. `y.md`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-snowflake/src/main/java/io.airbyte.integrations.source.snowflake/SnowflakeSource.java [hunk 1]
```diff
@@ -53,9 +53,4 @@ public JsonNode toDatabaseConfig(final JsonNode config) {
         "INFORMATION_SCHEMA");
   }
 
-  @Override
-  protected JdbcSourceOperations getSourceOperations() {
-    return new SnowflakeSourceOperations();
-  }
```
Comment (andriikorotkov) [inline airbyte-integrations/connectors/source-snowflake/src/main/java/io.airbyte.integrations.source.snowflake/SnowflakeSource.java:59]: Why do we need to remove `getSourceOperations` for Snowflake and MySQL? As far as I can see, the `putDouble` and `putBigInt` methods are overridden for the snowflake in the `SnowflakeSourceOperations` class, and the `putBoolean` method is overridden for `MySqlSourceOperations`. I think you need to revert the changes for snowflake and mysql and add those classes to the constructor.

**Review by andriikorotkov — APPROVED** [review]

**Review by yurii-bidiuk — APPROVED** [review]

**Review by alexandr-shegeda — APPROVED** [review]
LGTM. we can proceed with publish and merge without airbyte review

**Review by DoNotPanicUA — APPROVED** [review]

**Comment by sashaNeshcheret** [comment]
@tuliren Please, take a look the PR

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSourceOperations.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+public class MssqlSourceOperations extends JdbcSourceOperations {
+
+  protected void putBinary(final ObjectNode node, final String columnName, final ResultSet resultSet, final int index) throws SQLException {
+    byte[] bytes = resultSet.getBytes(index);
+    String value = new String(bytes);
+    node.put(columnName, value);
+  }
+
+}
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSourceOperations.java:19]: Nitpick. Add newline here.

File: airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceDatatypeTest.java [hunk 3]
```diff
@@ -210,7 +208,7 @@ protected void initTests() {
             .addExpectedValues("0001-01-01T00:00:00Z", "9999-12-31T00:00:00Z", null)
             .build());
 
-    // TODO JdbcUtils-> DATE_FORMAT is set as ""yyyy-MM-dd'T'HH:mm:ss'Z'"" for both Date and Time types.
+// TODO JdbcUtils-> DATE_FORMAT is set as ""yyyy-MM-dd'T'HH:mm:ss'Z'"" for both Date and Time types.
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceDatatypeTest.java:211]: Nitpick. The format change here is unnecessary.

**Review by tuliren — APPROVED** [review]
Looks good.

> Handled date, time, binary and bit data types for Mssql source and fixed appropriate test cases.

This PR only handles the binary type, while "date", "time", and "bit" types are not affected. Can you update the PR title to reflect that?

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSourceOperations.java (hunk 2 above)
Comment (sashaNeshcheret) [reply airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSourceOperations.java]: Thanks. Added

File: airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceDatatypeTest.java (hunk 3 above)
Comment (sashaNeshcheret) [reply airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceDatatypeTest.java]: Reverted. Thanks.

**Pull request merged**